File: examples-sdk/csharp/003-basic-hello-server/Main.cs
// 003-basic-hello-server — C# SDK equivalent
// Compile: vil compile --from csharp --input 003-basic-hello-server/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("vil-basic-hello-server", 8080);
var gw = new ServiceProcess("gw");
gw.Endpoint("POST", "/transform", "transform");
gw.Endpoint("POST", "/echo", "echo");
gw.Endpoint("GET", "/health", "health");
server.Service(gw);
server.Compile();

File: examples-sdk/kotlin/003-basic-hello-server/main.kt
// 003-basic-hello-server — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 003-basic-hello-server/main.kt --release

fun main() {
    val server = VilServer("vil-basic-hello-server", 8080)
    val gw = ServiceProcess("gw")
    gw.endpoint("POST", "/transform", "transform")
    gw.endpoint("POST", "/echo", "echo")
    gw.endpoint("GET", "/health", "health")
    server.service(gw)
    server.compile()
}
